### PR TITLE
Added CA support: signing and storing certificates

### PR DESCRIPTION
**Pull request by reinkrul**

### CODE REVIEW

File: pkg/client.go [hunk 1]
```diff
@@ -35,6 +37,10 @@ type Client interface {
 	GenerateKeyPairFor(legalEntity types.LegalEntity) error
 	// SignFor signs a piece of data for a legal entity
 	SignFor(data []byte, legalEntity types.LegalEntity) ([]byte, error)
+	// SignCertificate issues a certificate by signing a PKCS10 certificate request. The private key of the specified CA should be available in the key store.
+	SignCertificate(entity types.LegalEntity, ca types.LegalEntity, pkcs10 []byte, profile CertificateProfile) ([]byte, error)
+	// GetPrivateKey returns the current private key for a given legal entity. It can be used for signing, but cannot be exported.
+	GetPrivateKey(entity types.LegalEntity) (crypto.Signer, error)
```
Comment (woutslakhorst) [inline pkg/client.go]: Untill now private keys did not leave the crypto module, what's the use case?

File: pkg/client.go [hunk 2]
```diff
@@ -47,6 +53,16 @@ type Client interface {
 	KeyExistsFor(legalEntity types.LegalEntity) bool
 }
 
+// CertificateProfile is used to specify input parameters for certificate issuance.
+type CertificateProfile struct {
+	KeyUsage     x509.KeyUsage
+	IsCA         bool
+	// MaxPathLen is ignored is IsCa = false
```
Comment (woutslakhorst) [inline pkg/client.go]: typo

File: pkg/crypto.go [hunk 3]
```diff
@@ -73,11 +88,90 @@ type CryptoConfig struct {
...
-	Storage storage.Storage
-	//keyCache map[string]rsa.PrivateKey
+	Storage    storage.Storage
 	Config     CryptoConfig
 	configOnce sync.Once
 	configDone bool
+	_logger    *logrus.Entry
+}
+
+type key struct {
```
Comment (woutslakhorst) [inline pkg/crypto.go]: bit of a generic type description

File: pkg/util_test.go [hunk 4]
```diff
@@ -0,0 +1,163 @@
+/*
+ * Nuts crypto
+ * Copyright (C) 2019. Nuts community
```
Comment (woutslakhorst) [inline pkg/util_test.go:3]: 2020

Comment (reinkrul) [reply pkg/util_test.go:3]: Shouldn't we fix it in all files? If so, I'd suggest we fix it in a seperate PR. (if not, I'll fix it here)

File: pkg/client.go (hunk 1 above)
Comment (reinkrul) [reply pkg/client.go]: Because it is used by nuts-registry to sign a CSR. Go's CSR API requires an implementation of the Signer interface to sign the CSR.

Comment (reinkrul) [reply pkg/client.go]: Renamed it to GetOpaquePrivateKey after discussion with @stevenvegt, since that's how Go calls it.

**Review by woutslakhorst — APPROVED** [review]

**Pull request merged**